shared/src/test/diff/pretyper/ucs/patterns/Literals.mls

```diff
@@ -10,25 +10,91 @@ class Pair[A, B](x: A, y: B)
 //│ class Pair[A, B](x: A, y: B)
 
 fun f(x) = if x is Some(1) then true else false
-//│ fun f: (Object & ~#Some | Some[Eql[1]]) -> Bool
+//│ fun f: (Object & ~#Some | Some[Object]) -> Bool
+
+[f(Some(1)), f(None), f(Some(2)), f(Some(-1))]
+//│ [Bool, Bool, Bool, Bool]
+//│ res
+//│     = [ true, false, false, false ]
+
+fun f(x) = if x is Some(1) then true
+//│ fun f: Some[1] -> true
+
+fun f(x) = if x is
+  Some(1) then true
+  Some(2) then false
+//│ fun f: Some[1 | 2] -> Bool
+
+[f(Some(1)), f(Some(2))]
+//│ [Bool, Bool]
+//│ res
+//│     = [ true, false ]
 
 fun g(x) = if x then 1 else 2
 //│ fun g: Bool -> (1 | 2)
 
 :e
 fun test_must_be_boolean(x) = if 0 then 1 else 2
 //│ ╔══[ERROR] Type mismatch in type ascription:
-//│ ║  l.19: 	fun test_must_be_boolean(x) = if 0 then 1 else 2
+//│ ║  l.37: 	fun test_must_be_boolean(x) = if 0 then 1 else 2
 //│ ║        	                                 ^
 //│ ╙── integer literal of type `0` is not an instance of type `Bool`
 //│ fun test_must_be_boolean: anything -> (1 | 2)
 
 fun g(x) = if x is true then 1 else 2
-//│ fun g: Eql[true] -> (1 | 2)
+//│ fun g: Object -> (1 | 2)
+
+[g(true), g(false), g(None)]
+//│ [1 | 2, 1 | 2, 1 | 2]
+//│ res
+//│     = [ 1, 2, 2 ]
 
 fun g(x) = if x && true is true then 1 else 2
 //│ fun g: Bool -> (1 | 2)
 
 fun h(x) = if (x : Bool) then 1 else 2
 //│ fun h: Bool -> (1 | 2)
 
+// Currently post-processing cannot hanlde this case. The desugared term is not
+// perfect. Also, is the inferred type wrong?
```

Comment:
What's the desugared term?

Reply:
```
case x of {
  true => "true";
  _ => case x of {
    Some =>
      let ucs$args_x$Some = (Some).unapply(x,) in
      let value = (ucs$args_x$Some).0 in
      "Some";
    _ => case x of {
        0 => "zero"
    }
  }
}
```

Reply:
We should add a way of pretty-printing the code in the new syntax so we can easily try it out by copy-pasting it...

Reply:
Do you want me to do that? I'd happy to work on it. 😈

Reply:
Sure, if you feel like it!